lib/rules/disallow-trailing-whitespace.js

```diff
@@ -64,13 +64,70 @@ module.exports.prototype = {
 
     check: function(file, errors) {
         var ignoreEmptyLines = this._ignoreEmptyLines;
-
         var lines = file.getLines();
+
         for (var i = 0, l = lines.length; i < l; i++) {
+
             if (lines[i].match(/\s$/) && !(ignoreEmptyLines && lines[i].match(/^\s*$/))) {
-                errors.add('Illegal trailing whitespace', i + 1, lines[i].length);
+                var fixed = false;
+                var currentLineNumber = i + 1;
+                var startLineNumber;
+                var precendingToken;
+                var targetToken;
+
+                while (!precendingToken && currentLineNumber > 0) {
+                    precendingToken = file.getLastTokenOnLine(currentLineNumber, {includeComments: true});
+                    currentLineNumber--;
+                }
+
+                if (typeof precendingToken === 'undefined') {
```

Comment:
It will never be true. It could be written like ``` precendintToken === undefined ```, but I believe it's a better practice to use typeof.